src/Api/Persistent.hs

```diff
@@ -159,6 +160,14 @@ getSpaces = getDb dbSpaces
 getIdeas :: Persist [Idea]
 getIdeas = getDb dbIdeas
 
+-- | If idea space already exists, return 'False'.  Otherwise, create it and return 'True'.
+addIdeaSpace :: IdeaSpace -> Persist Bool
```

Comment:
This should has type `IdeaSpace -> Persist ()` to hide information, and thow an error on data duplication.